### DISCUSSION TITLE
Totems+ V1.5.0 / 1yr Anniversary Update

### DISCUSSION THREAD

**Discussion by ben-wals** [The-Iceburg/TotemsPlus#65] [Announcements]
![totemsplusbanner](https://user-images.githubusercontent.com/67003539/202871281-fa96dc25-6402-4876-bf64-fd4667859017.png)

Whoa we’re (almost) halfway there! That’s right we’re (almost) halfway to v2 & just to add to the excitement it’s TotemPlus’s 1 year anniversary. So, buckle in as this update is going to be a big one!

### **TotemsPlusNet**

The first announcement is probably the biggest one! We’re proud to now officially announce our plans to bring Totems+ to the internet! That’s right we’re committed to bringing the full capability of the desktop version of TotemsPlus all the way to the internet. Just like Totems+ you’ll be able to upload your own files and edit everything from sounds to names, lore and more! All before downloading the complete packs to your device, no program installation needed! We all super excited for this new chapter in Totem Pluses future but I’m sure as well this may raise a few questions so take a look at the brief Q & A below:

- When will TotemsPlusNet release?
`We plan to start development on TotemsPlusNet after we reach v2 on the desktop repository creating a functioning version on all of those features before we work on anything new. It should also be noted that next year is a big year for the Devs IRL so development won’t be “finished” for a while however you can at least expect a functioning BETA by this time next year so hopefully by our 2yr anniversary well have something to show you all!`

- Will development here end?
`Whilst the initial version of TotemsPlusNet is being developed you can expect to see a lack of activity from this project as we are only a team of 2. However, once we bring TotemsPlusNet up to a point of parity we hope to develop features in parallel with the desktop version you can appreciate that this will cause longer development times but we hope to release fully functioning versions of new features on both projects simultaneously.`

- How will version numbers work?
`TotemsPlusNet won’t immediately start on version 2 and in fact we aim to switch our versioning method from Sequential versioning to semantic versioning (MAJOR.MINOR.PATCH) which is more widely used across the industry. We aim to do this with TotemsPlus at some point as well however a more appropriate time will be chosen.`

- Where can I find TotemsPlusNet?
`You can expect to find TotemsPlusNet at “`[totemsplus.net]( https://the-iceburg.github.io/TotemsPlusNet)`” we’re currently using github pages to host our static coming-soon page which you can already view! But as we get into uploading files and custom URLs ect, we will be moving to a different provider.`

### **TotemsPlus v2**

TotemsPlus v2 is almost halfway into its development process and we’d like to take the time to explain making use of this table what’s already here and what you can expect to see in our full release of version 2. It should be noted that a lot of v2 is work that’s been done refactoring under the hood but moving into the later half we hope to bring a lot more fully polished feature.

|                | What's here: | What's Coming |
| --------- | -------------- | ---------------- |
| Features | #49, #35, #38 | #62, #53, #51, #41, #40, #34, #13, #7 |
| Bugs       | #31, #52, #48, #50, #54 | #46, so far... |

If at any point during the development process you want to see our progress you can check it on the GitHub milestone: [V2.0.0](https://github.com/The-Iceburg/TotemsPlus/milestone/2). We’ll also be posting regular updates on our twitter: [The Iceburg](https://twitter.com/theiceburg21) and we’ll be sure to let you know when v2 is ready to ship!

### **What's next for Totems+? ~ PROGRESS**

We thought that as were almost halfway to v2 we should provide an update for you and ourselves on the promises we made for v2!
-	Refactoring
As aforementioned refactoring has been by far our main focus in these last updates here’s a quick overview on what we’ve done to improve our code so far including the massive JSON update which removed a total of 828 unnecessary lines of code. We continue to learn everyday while developing this product so expect many more refactors to come
-	Bugfixes
As you may have hopefully seen in the table above, we have heavily prioritised the fixing of bugs with our current OPEN bugs totalling just 1 (which we plan to have fixed very soon 😊)
-	GitHub Skills
We like to think that our GitHub skills have come a long way. We’re in the process of building out or wiki, We’ve been making use of projects to categorise our features and bugs and we’re using GitHub pages for TotemsPlusNet!

With all that said here are some additional promises we’d like to add
-	Issue Tags
We’ve noticed our Issue tags are becoming less meaningful/specific so we’d like to step back and re-define that system for our project. Further more we’d love to see these tags interacting with our project planners
-	Testing Proccess
Currently our testing process when developing and releasing new features is limited. We plan to outline a “test protocol” before branches can be merged which I’m sure will eventually end up somewhere in the project (likely the README)

### **TotemsPlus Week**
We’re planning a Totems+ Week before the new year where we’ll be trying to develop as many features as possible bringing us as close as possible to v2 all in one 1 week. The date is TBC however keep an eye out for it as there will tonnes of activity across all our platforms

So that’s all for now we’re super excited for our development timeline for this project an we hope to bring you are visions as soon as humanly possible. Remember as always to bear with us during development and stay safe out there!

**Comment by Dockuin** [comment]
Poggers update my guy - I'm so surprised at how far we've come within this year.
Not only with this project but our knowledge of programming and overall skills - including branching off into separate web development projects.
